lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -80,278 +78,727 @@
+                                            name=name,
+                                            size=size_bytes,
+                                            uuid=str(uuid4()))
+
+    return disk_payload
+
+
+def update_boot_order(playbook_instance):
+    """Updates the boot order of an instance using the boot order information
+        provided in the Ansible playbook input.
+
+    Args:
+        playbook_instance (dict): The specified instance configuration from
+        the playbook input
+    """
+    boot_order_payload = get_full_boot_order_payload_for_playbook(
+        playbook_instance)
+    instance_uuid = get_instance_by_name(playbook_instance['name']).uuid
+
+    RESOURCES['update_app'].edit_boot_order(boot_order_payload, instance_uuid)
+
+
+def get_full_boot_order_payload_for_playbook(playbook_instance):
+    """Given the playbook input, generate a payload to update the boot order
+        for the created instance.
+
+    Args:
+        playbook_instance (dict): The specified instance configuration from
+        the playbook input
+
+    Returns:
+        ApiEditApplicationPayload: A payload object containing all the boot
+            order objects needed to perform the update operation.
+    """    
+    playbook_devices = {}
+    playbook_devices['disks'] = playbook_instance['disks']
+    playbook_devices['nics'] = playbook_instance['nics']
+
+    existing_instance = get_instance_by_name(playbook_instance['name'])
+    existing_instance_boot_devices = existing_instance.boot_order
+
+    new_boot_order = []
+
+    for boot_device in existing_instance_boot_devices:
+        new_boot_order_entry = get_new_boot_order_entry_for_device(
+            boot_device, playbook_devices)
+
+        boot_order_payload = get_boot_order_payload(new_boot_order_entry)
+        new_boot_order.append(boot_order_payload)
+
+    new_boot_order = sorted(new_boot_order, key=lambda payload: payload.order)
+    full_boot_order_payload = tacp.ApiEditApplicationPayload(
+        boot_order=new_boot_order)
+
+    return full_boot_order_payload
+
+
+def get_new_boot_order_entry_for_device(boot_device, playbook_devices):
+    """Generates a dict of values necessary to populate an ApiBootOrderPayload
+        object for updating the boot order.
+
+    Args:
+        boot_device (ApiBootOrderPayload): The preexisting boot device
+            in the instance in question.
+        playbook_devices (list): A list of the vNICs and disks as provided
+            in the Ansible playbook, especially indicating the desired boot
+            order.
+
+    Returns:
+        dict: [description]
+    """
+    new_boot_order_entry = {}
+
+    name = boot_device.name
+    new_boot_order_entry['name'] = name
+
+    if boot_device.vnic_uuid:
+        new_boot_order_entry['vnic_uuid'] = boot_device.vnic_uuid
+        new_boot_order_entry['disk_uuid'] = None
+        playbook_nic = [nic for nic in playbook_devices['nics']
```

Comment:
You can use generator expressions to be more efficient:
https://www.python.org/dev/peps/pep-0289/

This would look like:
```python
playbook_nic = next(nic for nic in playbook_devices['nics'] if nic['name'] == name])
```

Another option is using function style with `filter`.